### ISSUE TITLE
Generated URLs have wrong scheme when TLS is handled by reverse proxy

### ISSUE THREAD

**Issue by moritzsternemann** [LycheeOrg/Lychee#719]
### Detailed description of the problem [REQUIRED]
I have Lychee set-up behind a reverse proxy which handles TLS, so the Nginx instance actually serving the PHP app is only handling plain http. I think as a result the Laravel URL generation helper generates URLs starting with `http` instead of `https` for assets (see [Laravel Documentation](https://laravel.com/docs/8.x/urls#generating-basic-urls)).
These plain HTTP asset URLs are then blocked by the browser because of the CSP.

I temporarily resolved the issue by enabling the existing `URL::forceScheme('https');` statement [for example in the install route](https://github.com/LycheeOrg/Lychee/blob/98a90f3ad543dbfd78c4cf5e4bc965381cabe576/routes/install.php#L18).
However, I think this should ideally be configurable separately from the `APP_ENV`.

The configured `APP_URL` contains the `https` scheme.

### Steps to reproduce the issue
**Steps to reproduce the behavior:**
1. Serve Lychee from a web server without TLS
2. Reverse proxy that Lychee instance with TLS
3. Observe the generated asset URLs having the wrong scheme (`http` instead of `https`)

**Screenshots**
<img width="1648" alt="Screen Shot 2020-09-09 at 22 17 07" src="https://user-images.githubusercontent.com/3034168/92649091-cb948a00-f2ea-11ea-8efc-17311404e080.png">

### Output of the diagnostics  [REQUIRED]
*Errors and warnings are probably pretty irrelevant because the problem already occurrs while accessing the first page of the installer.*
```
Diagnostics
-----------
Warning: You may experience problems when uploading a large amount of photos. Take a look in the FAQ for details.
Warning: Pictures that are rotated lose their metadata! Please enable Imagick in settings to avoid that.
Error: username not set in database
Error: password not set in database
Error: sorting_Photos not set in database
Error: sorting_Albums not set in database
Error: imagick not set in database
Error: skip_duplicates not set in database
Error: check_for_updates not set in database
Error: version not set in database
Error: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'lychee.configs' doesn't exist (SQL: select `key`, `value`, `type_range` from `configs`)
Warning: Dropbox import not working. No property for dropbox_key.

System Information
------------------
Lychee Version (git):            master (6cb6386) - Data not in Cache
DB Version:                      4.0.0

composer install:                --no-dev
APP_ENV:                         production
APP_DEBUG:                       false

System:                          Linux
PHP Version:                     7.4
Max uploaded file size:          100M
Max post size:                   100M
MySQL Version:                   10.4.14-MariaDB

Imagick:                         1
Imagick Active:                  key not found in settings
Imagick Version:                 1691
GD Version:                      2.3.0

Config Information
------------------
Error: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'lychee.configs' doesn't exist (SQL: select * from `configs` where `confidentiality` <= 2 order by `id` asc)
```

**Comment by d7415** [comment]
~~For your stated issue: What have you set as your `APP_URL` in `.env`?~~

Aside: It looks like you haven't run `php artisan migrate` yet, which is part of the installation procedure - you may wish to check for any other missed steps.

**Comment by moritzsternemann** [comment]
@d7415 Sorry for the late response. I tried again using a new testing instance of lychee and found out that the problem indeed only occurs before running `php artisan migrate`.
During the initial installation I thought that the migrations are run as one of the last steps of the installer (`/install/migrate`), because running them explicitly isn't mentioned in the new documentation over at https://lycheeorg.github.io/docs/installation.html. Just now I found it exists in the old documentation: https://github.com/LycheeOrg/Lychee/wiki/Install#5-migrate-the-database.
Should we add this step to the new docs for clarification?

**Comment by d7415** [comment]
That's good.
I think the installer is meant to handle that.. @ildyria?

**Comment by ildyria** [comment]
https://github.com/LycheeOrg/Lychee/blob/9c65f672e1a06ef303a36a9029870f1dd3dc64cf/app/ControllerFunctions/Install/ApplyMigration.php#L32-L34

https://github.com/LycheeOrg/Lychee/blob/9c65f672e1a06ef303a36a9029870f1dd3dc64cf/app/Http/Controllers/Install/MigrationController.php#L23-L27

**TL;DR: Installer should apply the migration.**

**Comment by d7415** [comment]
In that case it would be nice to debug why it didn't, but we may be beyond that point.

@moritzsternemann If you have any clues as to what might have stopped it, we'd be interested to hear them. I'm going to close this for now as the original issue has been resolved, but I'm happy to reopen if there's more information to work from.

**Comment by girlpunk** [comment]
Hi, is manually running the migration still expected as a requirement for this to function correctly?

**Comment by d7415** [comment]
> Hi, is manually running the migration still expected as a requirement for this to function correctly?

It's been 3 years with no other reports, so no.
